### DISCUSSION TITLE
Optional path parameters

### DISCUSSION THREAD

**Discussion by Bird87ZA** [goadesign/goa#2937] [Q&A]
In our API we have a `GET` route that can either receive a `id` param or not.

If the `id` is passed, it will find all categories with that category as parent. If the `id` is not past, it will list all categories with `parent_id` = 1.

In the design, I have this:

```
HTTP(func() {
	Path("/categories/list")
	Cookie("session_id:ALTSESSID")
})
Method("CategoriesList", func() {
	Description("Get a list of categories.")
	Payload(CategoryListRequest)
	Result(CategoryListResponse)

	HTTP(func() {
		GET("{id}")
		Response(StatusOK)
		Response("unexpected", StatusInternalServerError)
		Response("upstream_unavailable", StatusServiceUnavailable)
		Response("timeout", StatusGatewayTimeout)
	})
})
```

How do I specify the `{id}` to be optional?

**Comment by raphael** [comment]
In this case you need to define two separate methods, one with the path "/{id}" and one with the path "/" - these can be implemented by calling the same method.
